mkdir: the argument '--parents' cannot be used multiple times

When building [c2rust](https://github.com/immunant/c2rust) on darwin, throw this error:

```
  make: warning: jobserver unavailable: using -j1.  Add `+' to parent make rule.
  Cloning into 'tinycbor_build'...
  HEAD is now at d393c16 Parser: fix reading it->extra on big endian when bytesNeeded == 1
  error: the argument '--parents' cannot be used multiple times

  Usage: mkdir [OPTION]... DIRECTORY...

  For more information, try '--help'.
  make[4]: *** [lib/libtinycbor.a] Error 1
  make[3]: *** [tinycbor/src/tinycbor_build-stamp/tinycbor_build-build] Error 2
  make[2]: *** [CMakeFiles/tinycbor_build.dir/all] Error 2
  make[1]: *** [CMakeFiles/clangAstExporter.dir/rule] Error 2
  make: *** [clangAstExporter] Error 2
```

> mkdir 0.0.27

--- a/src/uu/mkdir/src/mkdir.rs
+++ b/src/uu/mkdir/src/mkdir.rs
@@ -114,6 +114,7 @@ pub fn uu_app() -> Command {
                 .short('p')
                 .long(options::PARENTS)
                 .help("make parent directories as needed")
+                .overrides_with(options::PARENTS)
                 .action(ArgAction::SetTrue),
         )
         .arg(

--- a/tests/by-util/test_mkdir.rs
+++ b/tests/by-util/test_mkdir.rs
@@ -61,9 +61,21 @@ fn test_mkdir_parent() {
     let test_dir = "parent_dir/child_dir";
 
     scene.ucmd().arg("-p").arg(test_dir).succeeds();
-    scene.ucmd().arg("-p").arg(test_dir).succeeds();
+    scene.ucmd().arg("-p").arg("-p").arg(test_dir).succeeds();
     scene.ucmd().arg("--parent").arg(test_dir).succeeds();
+    scene
+        .ucmd()
+        .arg("--parent")
+        .arg("--parent")
+        .arg(test_dir)
+        .succeeds();
     scene.ucmd().arg("--parents").arg(test_dir).succeeds();
+    scene
+        .ucmd()
+        .arg("--parents")
+        .arg("--parents")
+        .arg(test_dir)
+        .succeeds();
 }
 
 #[test]

